docs/reference/hooks.rst

```diff
@@ -792,6 +792,21 @@ Hooks for customising the way users are directed through the process of creating
   The function does not have to return anything, but if an object with a ``status_code`` property is returned, Wagtail will use it as a response object and skip the rest of the view.
 
 
+.. _construct_synced_page_tree_list:
+
+``construct_synced_page_tree_list``
```

Comment:
The naming of this hook makes it sound generic when it's really just for translations. It also may imply that this is all that's needed to keep one page in sync with another.

I wonder if we should make it generic (so it could be used for hooking in pages for other use cases than translation), or name it in a way so that it's clear it's something only translation packages need to be aware of?

If the latter, I wonder if this has any use outside of ``wagtail_localize``/``wagtail_simple_translation`` so maybe we should consider not documenting it at all?

Reply:
I can see other potential uses, outside of `wagtail_localize`/`wagtail_simple_translation`, but perhaps it is best to change it to something like `construct_synced_translation_list` ? (any other names are welcome)

Reply:
sorted in [c71fafe](https://github.com/wagtail/wagtail/pull/7984/commits/c71fafedd674612d10f01084777a2e4a5ce67715) as per our discussion